compile error in gcc-6.0 and above

build ok. but checkall is failed.
I tested gcc-6.3.0 and gcc-7.1.0.

```
$ ./waf build
(many many warning... but no error)
$ ./waf --checkall
In file included from ../jubatus/util/lang/ref_test.cpp:33:0:                                                                                                                                                                                                                     [434/9990]
../.unittest-gtest/gtest-1.7.0/fused-src/gtest/gtest.h:1664:14: error: ‘tuple’ is already declared in this scope                     
 using ::std::tuple;                                                                                               
              ^~~~~                                                                                                                                                                                                                         
../.unittest-gtest/gtest-1.7.0/fused-src/gtest/gtest.h:1665:14: error: ‘tuple_element’ is already declared in this scope                        
 using ::std::tuple_element;                                                                                                                               
              ^~~~~~~~~~~~~                                                                                                                                                                                                      
../.unittest-gtest/gtest-1.7.0/fused-src/gtest/gtest.h:1666:14: error: ‘tuple_size’ is already declared in this scope                                                                 
 using ::std::tuple_size;                                                                                                            
              ^~~~~~~~~~                                                                                                                        
                                                                                                                                                               
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/include/g++-v6/tr1/functional:39:0,
                 from ../jubatus/util/lang/ref.h:37,                                                                                         
                 from ../jubatus/util/lang/mem_fn.h:35,                                                                                                                               
                 from ../jubatus/util/lang/bind.h:35,                                                                                
                 from ../jubatus/util/lang/bind_test.cpp:34:                                      
/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/include/g++-v6/tr1/tuple:130:11: error: redefinition of ‘class std::tuple< <template-parameter-1-1> >’
     class tuple : public _Tuple_impl<0, _Elements...>                                                                                          
           ^~~~~
In file included from ../.unittest-gtest/gtest-1.7.0/fused-src/gtest/gtest.h:1655:0,
                 from ../jubatus/util/lang/bind_test.cpp:32:
/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/include/g++-v6/tuple:554:11: error: previous definition of ‘class std::tuple< <template-parameter-1-1> >’
     class tuple : public _Tuple_impl<0, _Elements...>
           ^~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/include/g++-v6/tr1/functional:39:0,
                 from ../jubatus/util/lang/ref.h:37,
                 from ../jubatus/util/lang/mem_fn.h:35,
                 from ../jubatus/util/lang/bind.h:35,
                 from ../jubatus/util/lang/bind_test.cpp:34:
/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/include/g++-v6/tr1/tuple:164:20: error: redefinition of ‘class std::tuple<>’
   template<> class tuple<> { };
                    ^~~~~~~
In file included from ../.unittest-gtest/gtest-1.7.0/fused-src/gtest/gtest.h:1655:0,
                 from ../jubatus/util/lang/bind_test.cpp:32:
/usr/lib/gcc/x86_64-pc-linux-gnu/6.3.0/include/g++-v6/tuple:857:11: error: previous definition of ‘class std::tuple<>’
     class tuple<>
           ^~~~~~~
(repeats same errors)
```

--- a/jubatus/util/lang/enable_shared_from_this.h
+++ b/jubatus/util/lang/enable_shared_from_this.h
@@ -33,7 +33,7 @@
 #define JUBATUS_UTIL_LANG_ENABLE_SHARED_FROM_THIS_H_
 
 #include <memory>
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 #include <tr1/memory>
 #endif
 #include "shared_ptr.h"
@@ -43,7 +43,7 @@ namespace util {
 namespace lang {
 
 namespace detail {
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 namespace enable_shared_from_this_ns = ::std::tr1;
 #else
 namespace enable_shared_from_this_ns = ::std;

--- a/jubatus/util/lang/function.h
+++ b/jubatus/util/lang/function.h
@@ -33,7 +33,7 @@
 #define JUBATUS_UTIL_LANG_FUNCTION_H_
 
 #include <functional>
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 #include <tr1/functional>
 #endif
 
@@ -42,7 +42,7 @@ namespace util {
 namespace lang {
 
 namespace detail {
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 namespace function_ns = ::std::tr1;
 #else
 namespace function_ns = ::std;

--- a/jubatus/util/lang/ref.h
+++ b/jubatus/util/lang/ref.h
@@ -33,7 +33,7 @@
 #define JUBATUS_UTIL_LANG_REF_H_
 
 #include <functional>
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 #include <tr1/functional>
 #endif
 
@@ -42,7 +42,7 @@ namespace util {
 namespace lang {
 
 namespace detail {
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 namespace ref_ns = ::std::tr1;
 #else
 namespace ref_ns = ::std;

--- a/jubatus/util/lang/shared_ptr.h
+++ b/jubatus/util/lang/shared_ptr.h
@@ -34,7 +34,7 @@
 
 #include <exception>
 #include <memory>
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 #include <tr1/memory>
 #endif
 
@@ -64,7 +64,7 @@ template <class T>
 class enable_shared_from_this;
 
 namespace detail {
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 namespace shared_ptr_ns = ::std::tr1;
 #else
 namespace shared_ptr_ns = ::std;

--- a/jubatus/util/lang/weak_ptr.h
+++ b/jubatus/util/lang/weak_ptr.h
@@ -33,7 +33,7 @@
 #define JUBATUS_UTIL_LANG_WEAK_PTR_H_
 
 #include <memory>
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 #include <tr1/memory>
 #endif
 
@@ -42,7 +42,7 @@ namespace util {
 namespace lang {
 
 namespace detail {
-#ifdef __GLIBCXX__
+#if defined(__GLIBCXX__) && __cplusplus <= 199711L
 namespace weak_ptr_ns = ::std::tr1;
 #else
 namespace weak_ptr_ns = ::std;

